server/sandbox_remove.go

```diff
@@ -32,29 +29,16 @@ func (s *Server) RemovePodSandbox(ctx context.Context, req *pb.RemovePodSandboxR
 
 	// Delete all the containers in the sandbox
 	for _, c := range containers {
-		if err := s.runtime.UpdateStatus(c); err != nil {
-			return nil, fmt.Errorf("failed to update container state: %v", err)
-		}
-
-		cState := s.runtime.ContainerStatus(c)
-		if cState.Status == oci.ContainerStateCreated || cState.Status == oci.ContainerStateRunning {
-			if err := s.runtime.StopContainer(c, -1); err != nil {
-				// Assume container is already stopped
-				logrus.Warnf("failed to stop container %s: %v", c.Name(), err)
-			}
-		}
-
+		// At this point we assume containers are stopped _always_
+		// because at least one call to StopPodSandbox is always made
+		// before calling remove
```

Comment:
that's not true in practice, read https://github.com/kubernetes-incubator/cri-o/pull/530#issuecomment-302958270 - we need to better document this in the CRI proto files. This PR is now on hold.